### PR TITLE
ncm-nfs: Support PanFS mounts

### LINKED ISSUE #452: ncm-nfs: schema too restrictive to use for panfs
PanFS mounts are very similar to nfs mounts, so we should extend the component to support them.
As requested by @drossy

### PR DESCRIPTION
**Pull request by jrha**
Fixes #452 as requested by @drossy

### CODE REVIEW

**Comment by jrha** [comment]
Needs tests and an example in the documentation

**Comment by stdweird** [comment]
there is code in the component that checks for nfs and should be modified like `($fstype =~ /^nfs/)` (but the unittests will reveal that :smile: )

**Comment by jrha** [comment]
Hmm, there are only 00-load and pod-syntax tests present right now :sob:

**Comment by drossy** [comment]
Looks good to me, is it worth adding a comment to the documentation saying that extending the regexp is the way to go for others in a similar situation?

**Comment by jrha** [comment]
Good point!

File: ncm-nfs/src/main/pan/components/nfs/schema.pan [hunk 1]
```diff
@@ -15,7 +15,7 @@ type structure_nfs_exports = {
 type structure_nfs_mounts = {
     'device'     : string
     'mountpoint' : string
-    'fstype'     : string with match(SELF, '^nfs|none$')
+    'fstype'     : string with match(SELF, '^nfs|panfs|none$')
```
Comment (stdweird) [inline ncm-nfs/src/main/pan/components/nfs/schema.pan]: you are aware that the string `"nfspanfsnone"` matches this regexp, right :smile: use
```pan
match(SELF, '^(nfs|panfs|none)$')
```

(pan really needs a `in` test)

File: ncm-nfs/src/main/perl/nfs.pm [hunk 2]
```diff
@@ -114,7 +114,7 @@ sub Configure($$@) {
       $freq = 0 unless defined($freq);
       $passno = 0 unless defined($passno);
       
-      if ( ($fstype =~ /^nfs/) || (($fstype eq 'none') && ($opt eq 'bind')) ) {
+      if ( ($fstype =~ /^(pa)?nfs/) || (($fstype eq 'none') && ($opt eq 'bind')) ) {
```
Comment (stdweird) [inline ncm-nfs/src/main/perl/nfs.pm]: use `/^(pa)?nfs$/` ?

**Comment by stdweird** [comment]
@jrha minor regexp remarks (i prefer strict regexps)

File: ncm-nfs/src/main/perl/nfs.pm (hunk 2 above)
Comment (jrha) [inline ncm-nfs/src/main/perl/nfs.pm]: Pffft, blame who ever wrote it first :stuck_out_tongue:

File: ncm-nfs/src/main/pan/components/nfs/schema.pan (hunk 1 above)
Comment (jrha) [inline ncm-nfs/src/main/pan/components/nfs/schema.pan]: Ugh, well so does `nfsbanana`

Comment (jrha) [inline ncm-nfs/src/main/pan/components/nfs/schema.pan]: But then nfsbanana and elephantnone would have matched the old regexp.

Comment (stdweird) [inline ncm-nfs/src/main/pan/components/nfs/schema.pan]: i didn't mean to imply that you introduced the support for nfsbanana, but might as well fix it :smile:

Comment (jrha) [inline ncm-nfs/src/main/pan/components/nfs/schema.pan]: :smiling_imp: understood!

**Comment by jrha** [comment]
Is something wrong with Jenkins?

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
@jrha disk full, should be ok again

**Comment by stdweird** [comment]
test this please

**Comment by jrha** [comment]
Yeah it looks sick. :hankey:

**Comment by jrha** [comment]
Yay! :smile:

**Comment by drossy** [comment]
nfs4 is a valid filesystem type that would have been accepted by the old regexp (and I think I saw a mention of it somewhere in the quattor code but can't find it right now)

**Comment by stdweird** [comment]
@drossy true, then it should be allowed in the schema and the perl regexp like
```pan
match(SELF, '^(nfs4?|panfs|none)$')
```
and 
```perl
$fstype =~ /^(nfs4?|panfs)$/
```

**Comment by jrha** [comment]
@stdweird happy?

**Comment by stdweird** [comment]
@jrha LGTM

**Pull request merged**